typeql-src/modules/ROOT/pages/fundamentals.adoc

```diff
@@ -396,36 +400,100 @@ For example, given the schema:
-  relates group-member;
+    relates user-group,
+    relates group-member;
 
 user-group sub entity
-  plays group-membership:user-group;
+    plays group-membership:user-group;
 
 subject sub entity,
-  plays group-membership:group-member;
+    plays group-membership:group-member;
 user sub subject;
 ----
 
 `group-membership` is a `relation` type that defines `user-group` and `group-member` roles.
 The `user-group` role is to be played by a `user-group` entity type whereas the `group-member` role is to be played by
 a `subject` type and all its subtypes entities.
 ====
+// end::relation_type_examples[]
+
+==== Roles
+
+Roles are special internal types used by relations. We can't create an instance of a role in a database.
+But we can set an instance of another type (role player) to play a role in a particular instance of a relation type.
 
 Roles allow a schema to enforce logical constraints on types of role players.
 
-.See the example of a role type constraint
+.See the example of a role player's type constraint
 [%collapsible]
 ====
-For example, a `group-membership` relation cannot associate a `user` type entity with a `file` type entity, because
-`file` type entity can't play any role in a `group-membership` relation.
+For example, given the schema below, a `file` type entity can't play any role in a `group-membership` relation,
+and a `user` type entity can play the `group-member` role of a `group-membership` relation because it inherits it
+from the `subject` type.
+
+[,typeql]
+----
+define
+
+group-membership sub relation,
+    relates user-group,
+    relates group-member;
+
+user-group sub entity
+    plays group-membership:user-group;
+
+subject sub entity,
+    plays group-membership:group-member;
+user sub subject;
+
+file sub entity;
+----
 ====
 
-// end::relation_type_examples[]
+==== Abstract roles
 
-[NOTE]
+[IMPORTANT]
+====
+Since TypeDB v.2.22.0, all roles are concrete, even in <<_abstract_types,abstract>> relations.
 ====
-Roles can be used in queries as a part of a relation even though roles do not have a direct type definition.
+
+.See the example of a concrete role in abstract type
+[%collapsible]
+====
+For example, given the following schema:
+
+[,typeql]
+----
+define
+
+membership sub relation,
+    abstract,
+    relates member;
+
+group-membership sub membership;
+----
+
+The concrete relation type `group-membership` inherits the `member` role from the abstract relation
+type `membership`.
+
+Previously, that would make the role `membership` to be abstract. To insert an instance
+of the `group-membership` type, we would have to override the abstract role with a concrete one with a new name.
+
+Now, we can use the role `membership` directly without the need to override it with a new role:
+
+[,typeql]
+----
+insert
+(member: $x) isa group-membership;
+----
+====
+
+[WARNING]
+====
+Old databases created in previous versions of TypeDB can retain the old behaviour with possible abstract roles.
```

Comment:
Schemas in databases created in versions of TypeDB prior to 2.22.0 will continue to use abstract roles when originally defined in abstract relations.

Reply:
It seems wrong to me to talk about schema in this context - as if a schema is using abstract roles. 
On top of that, schema, if we export it in TypeQL, would be exactly the same in both versions (prior to 2.22.0 and since 2.22.0). I know it's in the implementation that we can't see in TypeQL. 
So, it seems natural to talk about database behavior instead of schema here to avoid explaining particular details.

Reply:
Fixed in https://github.com/vaticle/typedb-docs/pull/714/commits/303e68ed2aff65463035d2205890d009ff6a06d8